### ISSUE TITLE
Different behavior of workspace_layout option from i3

### ISSUE THREAD

**Issue by lazy-dolphin** [swaywm/sway#5157]
Running sway 1.4_1 on Void Linux.
One of the features which I was used to in i3 was the ability to set the default workspace layout as stacking/tabbed because I only need tiling when I need to look at two or more windows at once. However, while using sway I encountered several different behavior which cause inconvenience while using this feature.

## Steps to reproduce:
1. Set 'workspace_layout stacking' in config.
2. Starting from a fresh workspace, spawn two or more windows in the same stack.
3. Move one of the windows right of the screen.
4. Move the same window left.
5. Close all windows.
6. Spawn multiple windows.

## What happens in Sway:
After moving the window left (step 4),
* In case two windows were spawned in step 2, the window on the right swaps positions with the window on the left. (Unexpected behavior)
* In case more than two windows were spawned in step 2, the window on the right gets merged back into the window stack on the left.

After spawning multiple windows (step 6), the windows spawn in vertical split. (Unexpected behavior)

## What happens in i3/What should happen in sway:
After moving the window left (step 4), the window on the right always gets merged back into the window stack on the left.

After spawning multiple windows (step 6), the windows spawn in stacking mode.

**Comment by wentasah** [comment]
I confirm that. Below are the differences between window trees in i3 and sway. The trees were obtained with:

    swaymsg -t get_tree|jq 'def f: {name, type, layout, nodes: [.nodes[]|f]}; recurse(.nodes[])|select(.name == "1")|f'

The i3 and sway config files contained `workspace_layout tabbed`.

In i3, a workspace with two shell windows in tabbed layout produces the following:
```json
{
	"name": "1",
	"type": "workspace",
	"layout": "splith",
	"nodes": [
		{
			"name": null,
			"type": "con",
			"layout": "tabbed",
			"nodes": [
				{
					"name": "shell: ~",
					"type": "con",
					"layout": "splith",
					"nodes": []
				},
				{
					"name": "shell: ~",
					"type": "con",
					"layout": "splith",
					"nodes": []
				}
			]
		}
	]
}
```
whereas in sway:
```json
{
	"name": "1",
	"type": "workspace",
	"layout": "tabbed",
	"nodes": [
		{
			"name": "shell: ~",
			"type": "con",
			"layout": "none",
			"nodes": []
		},
		{
			"name": "shell: ~",
			"type": "con",
			"layout": "none",
			"nodes": []
		}
	]
}
```

After moving one window to the right, i3 gives:
```json
{
	"name": "1",
	"type": "workspace",
	"layout": "splith",
	"nodes": [
		{
			"name": null,
			"type": "con",
			"layout": "tabbed",
			"nodes": [
				{
					"name": "shell: ~",
					"type": "con",
					"layout": "splith",
					"nodes": []
				}
			]
		},
		{
			"name": null,
			"type": "con",
			"layout": "tabbed",
			"nodes": [
				{
					"name": "shell: ~",
					"type": "con",
					"layout": "splith",
					"nodes": []
				}
			]
		}
	]
}
```
whereas sway gives:
```json
{
	"name": "1",
	"type": "workspace",
	"layout": "splith",
	"nodes": [
		{
			"name": "shell: ~",
			"type": "con",
			"layout": "none",
			"nodes": []
		},
		{
			"name": null,
			"type": "con",
			"layout": "tabbed",
			"nodes": [
				{
					"name": "shell: ~",
					"type": "con",
					"layout": "none",
					"nodes": []
				}
			]
		}
	]
}
```
In summary:

- i3 windows are grandchildren of a workspace, with an intermediate container. Sway allows windows to be direct children of a workspace.
- After moving the windows side-by-side, i3 tree structure is symmetric with both windows being a child of one of two intermediate containers. In sway, the result is not symmetric; only one window is placed in the intermediate container. The other window remains direct child of the workspace.
